packages/runtime-core/src/helpers/resolveAssets.ts

```diff
@@ -144,3 +153,27 @@ function resolve(registry: Record<string, any> | undefined, name: string) {
       registry[capitalize(camelize(name))])
   )
 }
+
+/**
+ * @private
+ */
+export function resolveLateAddedTag(
+  name: string,
+  key: 'setupState' | 'props',
+): unknown {
+  if (!currentRenderingInstance || !currentRenderingInstance[key]) return name
+  const data = currentRenderingInstance[key]
+  const value = data[name]
+  // Only the render function for the value is parsed as a component
+  // and a warning is reported
+  if (
+    __DEV__ &&
+    value &&
+    (value as ComponentInternalInstance).render &&
```

Comment:
In the 'setup script', the component value comes from the 'props' or 'setupState', and when it is an object with a render function, can it be considered a component? If it can be judged that it is a component in this way, then report a warning, otherwise leave it as is